runtime/src/dpe_crypto.rs

```diff
@@ -281,21 +281,41 @@ impl<'a> Crypto for DpeCrypto<'a> {
         match algs {
             AlgLen::Bit256 => Err(CryptoError::Size),
             AlgLen::Bit384 => {
+                // derive an EC key pair from the CDI
+                // note: the output point must be kept secret since it is derived from the private key,
+                // so as long as that output is kept secret and not released outside of Caliptra,
+                // it is safe to use it as key material.
+                let (_, hmac_seed) = Self::derive_key_pair(self, algs, cdi, label, info)?;
+
+                // create ikm to the hmac kdf by hashing the seed entropy from the pub key
+                // this is more secure than directly using the pub key components in the hmac
+                // kdf since the distribution of the pub key is taken from a set of discrete points
+                let mut hasher = Self::hash_initialize(self, algs)?;
+                hasher.update(hmac_seed.x.bytes())?;
+                hasher.update(hmac_seed.y.bytes())?;
+                let hmac_ikm: [u8; 48] = hasher
+                    .finish()?
+                    .bytes()
+                    .try_into()
+                    .map_err(|_| CryptoError::Size)?;
+
+                // derive an hmac key
+                let mut hmac_key = Array4x12::default();
                 hmac384_kdf(
                     self.hmac384,
-                    KeyReadArgs::new(*cdi).into(),
-                    label,
-                    Some(info),
+                    Hmac384Key::Array4x12(&Array4x12::from(hmac_ikm)),
+                    &[],
+                    None,
                     self.trng,
-                    KeyWriteArgs::new(KEY_ID_DPE_PRIV_KEY, KeyUsage::default().set_hmac_key_en())
-                        .into(),
+                    Hmac384Tag::Array4x12(&mut hmac_key),
                 )
                 .map_err(|e| CryptoError::CryptoLibError(u32::from(e)))?;
 
+                // sign digest with HMAC key
                 let mut tag = Array4x12::default();
                 self.hmac384
                     .hmac(
-                        &KeyReadArgs::new(KEY_ID_DPE_PRIV_KEY).into(),
+                        &Hmac384Key::Array4x12(&hmac_key),
```

Comment:
Done, https://github.com/chipsalliance/caliptra-dpe/pull/270 will zeroize hmac_seed, hmac_key is already being zeroized since Zeroize is derived for Array4xN, and I am now calling zeroize on hmac_ikm.